File: palindrome_number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let mut x_reversed = String::new();
        let x_string: String = x.to_string();

        for char in x_string.chars().rev() {
          x_reversed.push(char)
        }

        for (i,char) in x_string.char_indices() {
           if char != x_reversed.chars().nth(i).unwrap_or_default() {return false}
        }
        return true
    }

}

File: palindrome_number.js
const isPalindrome = (x) => {
	var xReversed = new String();
	const xAsString = x.toString();

	for (let char = xAsString.length - 1; char >= 0; char--) {
		xReversed += xAsString[char];
	}
	
	for (let char = 0; char < xReversed.length; char++) {
		if (xAsString[char] !== xReversed[char]){
			return false;
		}

	}
	return true
}

console.log(isPalindrome(-121))

